### PR TITLE
doc(azure): add azure ad docs

### PR DESCRIPTION
**Pull request by noahgorstein**
Adds documentation for the Azure AD integration with On-Prem

- fixes CLOUD-535

### CODE REVIEW

File: azure/README.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
++--------+
+```
+
+![Azure AD User Roles](./img/azure-user-groups.png)
+
+## How This Works
+
+1. A user clicks the "Sign in with Microsoft" button during login
+
+2. If the user successfully authenticates they are redirected the On-Prem home page where they can enter the Stardog Apps.
```
Comment (paulplace) [inline azure/README.md:24]: s/redirected the/redirected to the/

**Review by paulplace — APPROVED** [review]
Looks good. I had one minor wording suggestion.

What about configuring the Stardog server to also accept HTTP API calls (similar to what we did with Keycloak)? Did you try that? If not, it might be easier to try that now, while you still have your dev environment around.

File: azure/README.md (hunk 1 above)
Comment (noahgorstein) [reply azure/README.md]: fixed the grammar - as far as authenticating using a bearer token, this info is already contained in the Stardog docs: https://docs.stardog.com/operating-stardog/security/oauth-integration#azure-active-directory. This On-Prem integration does not require the claim containing the roles/groups to be formatted a specific way like Keycloak. Instead, this integration gets a user's group memberships via a Microsoft API call and then does what it needs to assemble the claim that is included in the JWT that is then sent to Stardog. 

Group membership is done using something like:
```python
    resp = client.get(
        "https://graph.microsoft.com/v1.0/me/transitiveMemberOf", token=token
    )
```

In short, this integration should not interfere.
